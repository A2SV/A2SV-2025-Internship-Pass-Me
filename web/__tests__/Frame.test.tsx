import React from "react";
import { render, screen } from "@testing-library/react";
import Frame from "../src/app/components/frame/frame";

// Mock next/image
jest.mock("next/image", () => ({
  __esModule: true,
  default: ({
    src,
    alt,
    width,
    height,
    className,
    priority,
    style,
  }: {
    src: string;
    alt: string;
    width?: number | string;
    height?: number | string;
    className?: string;
    priority?: boolean;
    style?: React.CSSProperties;
  }) => (
    // eslint-disable-next-line @next/next/no-img-element
    <img
      src={src}
      alt={alt}
      width={width}
      height={height}
      className={className}
      data-priority={priority?.toString()}
      style={style}
    />
  ),
}));

describe("Frame Component", () => {
  it("renders correctly with all elements", () => {
    render(<Frame />);

    // Verify banner image
    const banner = screen.getByAltText("A2SV Translator Banner");
    expect(banner).toBeInTheDocument();
    expect(banner).toHaveAttribute("src", "/banner.png");
    expect(banner).toHaveAttribute("width", "333");
    expect(banner).toHaveAttribute("height", "62");
    expect(banner).toHaveClass("md:w-[400px] md:h-auto object-contain");
    expect(banner).toHaveAttribute("data-priority", "true");

    // Verify message icon
    const messageIcon = screen.getByAltText("Messages icon");
    expect(messageIcon).toBeInTheDocument();
    expect(messageIcon).toHaveAttribute("src", "/messages.png");
    expect(messageIcon).toHaveAttribute("width", "48");
    expect(messageIcon).toHaveAttribute("height", "48");
    expect(messageIcon.parentElement).toHaveClass(
      "w-[48px] h-[48px] rounded-[8px] mb-8"
    );
    expect(messageIcon).toHaveStyle("filter: brightness(0) invert(1)");

    // Verify text content
    const text = screen.getByText(
      /Translate anything, anytime, with ease. A2SV Translator breaks language barriers/
    );
    expect(text).toBeInTheDocument();
    expect(text).toHaveClass(
      "font-normal text-white text-center max-w-[617px] text-[22px] leading-[30px]"
    );
  });

  it("has correct layout structure", () => {
    render(<Frame />);

    // Main container
    const mainContainer = screen.getByTestId("frame-container");
    expect(mainContainer).toHaveClass(
      "flex flex-col h-full w-full bg-[#1A1A1A] text-white"
    );

    // Banner container
    const bannerContainer = mainContainer.firstChild;
    expect(bannerContainer).toHaveClass("w-full flex justify-center py-4");

    // Content container
    const contentContainer = mainContainer.lastChild;
    expect(contentContainer).toHaveClass(
      "flex-1 flex flex-col items-center justify-center"
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Frame />);
    expect(asFragment()).toMatchSnapshot();
  });
});
