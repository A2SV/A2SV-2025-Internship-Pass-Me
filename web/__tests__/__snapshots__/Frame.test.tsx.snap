// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col h-full w-full bg-[#1A1A1A] text-white"
  >
    <div
      class="w-full flex justify-center py-4"
    >
      <img
        alt="A2SV Translator Banner"
        class="md:w-[400px] md:h-auto object-contain"
        data-priority="true"
        height="62"
        src="/banner.png"
        width="333"
      />
    </div>
    <div
      class="flex-1 flex flex-col items-center justify-center"
    >
      <div
        class="relative overflow-hidden w-[48px] h-[48px] rounded-[8px] mb-8"
      >
        <img
          alt="Messages icon"
          class="object-cover"
          height="48"
          src="/messages.png"
          style="filter: brightness(0) invert(1);"
          width="48"
        />
      </div>
      <p
        class="font-normal text-white text-center max-w-[617px] text-[22px] leading-[30px]"
      >
        Translate anything, anytime, with ease. A2SV Translator breaks language barriers, making communication smooth, fast, and accessible for everyone.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
